java/src/main/java/ai/rapids/cudf/Cuda.java

```diff
@@ -114,7 +114,7 @@ public String toString() {
     }
 
     @Override
-    public void close() {
+    public synchronized void close() {
```

Comment:
Why do we need to synchronize closing a stream, but none of the other classes?

Reply:
Most of the other classes synchronized on close, except `Stream` and `Event`. As of this PR, these don't synchronize: 

- `HostColumnVectorCore` (as far as as I can tell its `close` is overwritten by subclasses)
- nvcomp: `BatchedMetadata` and `Metadata`. Neither of these are called in a multi-threaded context.

I think we can synchronize close for the nvcomp ones, that seems like it could be a miss later. If you think this is unnecessary let me know and I can revert parts of this PR.

Reply:
> - `HostColumnVectorCore` (as far as as I can tell its `close` is overwritten by subclasses)

Child columns are instances of this class. So we should probably add `synchronized` to the close in `HostColumnVectorCore`.

Reply:
@revans2 did you want me to make a change around this?

Reply:
Please do

Reply:
ok done. I need to re-run my tests after upmerging. FYI.